File: spotify-backend/music-service/src/main/java/com/spotify/musicservice/dto/Track.java
package com.spotify.musicservice.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class Track {

    @JsonProperty("album")
    private Albums albums;

    @JsonProperty("artists")
    private List<Artist> artists;

    @JsonProperty("external_urls")
    private ExternalUrls externalUrls;

    @JsonProperty("id")
    private String id;

    @JsonProperty("name")
    private String name;

    @JsonProperty("release_date")
    private String releaseDate;

    @JsonProperty("preview_url")
    private String previewUrl;

    @JsonProperty("duration_ms")
    private long durationMs;

    @JsonProperty("type")
    private String type;

}

File: spotify-app-frontend/src/app/model/Track.ts
import { Album } from './Album';
import { Artist } from './Artist';
import { ExternalUrls } from './ExternalUrls';

export class Track{
    album!: Album;
    artists!: Artist[];
    ExternalUrls!: ExternalUrls;
    id!: string;
    name!: string;
    previewUrl!: string;
    type!: string;
    durationMs!: number;
    /*
    constructor(
        public album: Album,
        public artists: Artist[],
        public ExternalUrls: ExternalUrls,
        public id: string,
        public name: string,
        public previewUrl: string,
        public type: string
    ) {}
    */
}